config.gradle

```diff
@@ -9,7 +9,7 @@ def windowsReleaseLinkerArgs = [ '/OPT:REF', '/OPT:ICF' ]
 
 def linuxCompilerArgs = ['-std=c++1y', '-Wformat=2', '-Wall', '-Wextra', '-Werror', '-pedantic', '-Wno-psabi', '-g',
                          '-Wno-unused-parameter', '-fPIC', '-rdynamic', '-Wno-error=deprecated-declarations', '-pthread']
-def linuxLinkerArgs = ['-rdynamic', '-pthread']
+def linuxLinkerArgs = ['-rdynamic', '-pthread', '-latomic']

```

Comment:
Why is this needed?  We use std::atomic elsewhere already without this...

Reply:
https://stackoverflow.com/questions/30591313/why-does-g-still-require-latomic

From what I understand, it is because it is not a standard size and/or because it is not packed.  I don't see that as being too big of a deal, especially since it passes the unit tests I made

Reply:
Oh!  Yeah, there's no point in making a struct like that atomic.. all that does is make the lock implicit (the library is going to create a mutex).  Generally the largest lock-free atomics are 32 or 64 bit.  In this case it likely makes more sense to keep the lock then?

Reply:
If we want a lock-free way of doing this, I think the right way is to make it an atomic pointer and have the writing side atomically exchange the pointer to the new contents.

Reply:
I agree we should be doing pointer swaps for this. Once that happens, and the references in c functions get removed, I'll be good with this.